src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MethodCallHandler.java

```diff
@@ -115,17 +115,22 @@ private static boolean areMethodsChained(DetailAST ast1, DetailAST ast2) {
      * @return true if chained class are wrapped
      */
     private boolean isChainedMethodCallWrapped() {
-        boolean result = false;
         final DetailAST main = getMainAst();
         final DetailAST dot = main.getFirstChild();
-        final DetailAST target = dot.getFirstChild();
+        DetailAST target = dot.getFirstChild();
+        while (target.getType() == TokenTypes.INDEX_OP) {
+            target = target.getFirstChild();
+        }
 
         final DetailAST dot1 = target.getFirstChild();
-        final DetailAST target1 = dot1.getFirstChild();
+        boolean result = false;
 
-        if (dot1.getType() == TokenTypes.DOT
-            && target1.getType() == TokenTypes.METHOD_CALL) {
-            result = true;
+        if (dot1.getType() == TokenTypes.DOT) {
+            DetailAST target1 = dot1.getFirstChild();
```

Comment:
Not totally related to all your changes, but generally speaking, what is the `target` ast, and what is the significance of naming like dot/dot1 and targe/ target1?

Reply:
The whole code is quite confusing and missing many elements, I am working on this module for fixing other issues as well, wasted a few hours debugging cause I used `target` instead of `target1`, I have changed it in local will push to remote in a few.

Reply:
Please feel free to send minor commit with better variable names, current naming makes this method difficult to follow.